src/components/checkbox/checkbox.js

```diff
@@ -168,10 +168,19 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
 
       function keypressHandler(ev) {
         var keyCode = ev.which || ev.keyCode;
-        if (keyCode === $mdConstant.KEY_CODE.SPACE || keyCode === $mdConstant.KEY_CODE.ENTER) {
-          ev.preventDefault();
-          element.addClass('md-focused');
-          listener(ev);
+        ev.preventDefault();
+        switch(keyCode) {
+          case $mdConstant.KEY_CODE.SPACE:
+            element.addClass('md-focused');
+            listener(ev);
+            break;
+          case $mdConstant.KEY_CODE.ENTER:
+            var form = $mdUtil.getClosest(ev.target, 'form');
+            //We have to use a native event here as the form directive does not use jqlite
+            if(form) {
+              form.dispatchEvent(new Event('submit'));
```

Comment:
Dispatching an event is necessary. I believe a submit event is not dispatched by default because mdChecbox is not a native form control. If you move ev.preventDefault to only the SPACE case and remove the enter functionality the test for ensuring the form has been submitted will fail.